providers/mimetypes.rb

```diff
@@ -0,0 +1,68 @@
+#
+## Cookbook Name:: gecos-ws-mgmt
+## Provider:: pdf_viewer
+##
+## Copyright 2013, Junta de Andalucia
+## http://www.juntadeandalucia.es/
+##
+## All rights reserved - EUPL License V 1.1
+## http://www.osor.eu/eupl
+##
+
+require 'etc'
+require 'chef/mixin/shell_out'
+include Chef::Mixin::ShellOut
+
+action :setup do
+	begin
+	  if new_resource.support_os.include?($gecos_os)
+    	package "xdg-utils" do
+       action :nothing
+      end.run_action(:install)
+
+      users = new_resource.users
+
+      users.each_key do |user_key|
+        nameuser = user_key
+        username = nameuser.gsub('###','.')
+        user = users[user_key]
+				gid = Etc.getpwnam(username).gid
+				xdg_code = ''
+
+				directory "/home/#{username}/.local/share/applications" do
+					owner username
+					group gid
+					recursive true
+				end
+
+				Chef::Log.debug("mimetypes.rb - Users: #{user}")
+
+				user.mimetyperelationship.each do |assoc|
```

Comment:
Eso exige una comprobación de  cada uno de los tipos mimes configurados en
la receta.

Por ejemplo, tenemos este receta:

"mimetyperelationship": [
 {
   "mimetypes": [
      "application/pdf",
      "application/x-pdf",
      "application/x-aim"
   ],
   "desktop_entry": "evince.desktop"
 }


Habrá que lanzar una consulta como esta por cada uno de los tipos mime ahí
presentes:

xdg-mime query default application/pdf
xdg-mime query default application/x-pdf
xdg-mime query default application/x-aim

y comprobar si el resultado es evince.desktop.

Entiendo que es  eso lo que quereis¿?

Saludos.

2016-08-02 8:37 GMT+02:00 Alfonso E.M. <notifications@github.com>:

> In providers/mimetypes.rb
> <https://github.com/gecos-team/gecos-workstation-management-cookbook/pull/145#discussion_r73101139>
> :
>
> > +      users.each_key do |user_key|
> > +        nameuser = user_key
> > +        username = nameuser.gsub('###','.')
> > +        user = users[user_key]
> > +				gid = Etc.getpwnam(username).gid
> > +				xdg_code = ''
> > +
> > +				directory "/home/#{username}/.local/share/applications" do
> > +					owner username
> > +					group gid
> > +					recursive true
> > +				end
> > +
> > +				Chef::Log.debug("mimetypes.rb - Users: #{user}")
> > +
> > +				user.mimetyperelationship.each do |assoc|
>
> Como norma general las recetas deben comprobar lo que van a cambiar y no
> hacer nada si ya está cambiado. Ten en cuenta que las recetas tratan con un
> puesto de trabajo ocupado por un humano. Cuanto menos toques su disco y su
> cpu, mejor. Tanto para el disco como para el humano.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/gecos-team/gecos-workstation-management-cookbook/pull/145/files/c25d460433d1f0d07780a75a6172e8a068ae51dd#r73101139>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ATHDTAW9HipahiP-xuZeK_GFnULDt_0Eks5qbuWNgaJpZM4JRrKY>
> .
>